src/CodeGen_Internal.cpp

```diff
@@ -474,5 +474,31 @@ void set_function_attributes_for_target(llvm::Function *fn, Target t) {
     fn->addFnAttr("reciprocal-estimates", "none");
 }
 
+Expr compile_unsafe_promises(const Call *op, bool check_unsafe_promises) {
+    Expr result;
+
+    if (op->is_intrinsic(Call::unsafe_promise_clamped)) {
```

Comment:
Should this be an `internal_assert`? I'm a bit concerned this returns an undefined `Expr` if the call isn't a `unsafe_promise_clamped`.